### PR TITLE
Add support for column groups in tables

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This will be used in new memory tooling that @bkonyi is working on. This screenshot is example usage of this feature and these groups will not actually be added to the logging page.

Screen Shot 2022-06-23 at 12 25 50 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/table.dart [hunk 1]
```diff
@@ -1503,6 +1618,100 @@ class _TableRowState<T> extends State<TableRow<T>>
...
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(
+        horizontal: _columnGroupSpacingWithPadding - _columnGroupSpacing,
+      ),
+      child: Container(
+        width: _columnGroupSpacing,
+        decoration: BoxDecoration(
+          gradient: LinearGradient(
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/table.dart:1686]: Oh cool, I didn't know this was this simple!

File: packages/devtools_app/lib/src/shared/table_data.dart [hunk 2]
```diff
@@ -97,3 +98,12 @@ enum ColumnAlignment {
   right,
   center,
 }
+
+class ColumnGroup {
+  ColumnGroup({required this.name, required this.range});
+
+  final String name;
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/table_data.dart:105]: Nit: maybe `title`?

File: packages/devtools_app/lib/src/shared/table_data.dart [hunk 3]
```diff
@@ -97,3 +98,12 @@ enum ColumnAlignment {
   right,
   center,
 }
+
+class ColumnGroup {
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/shared/table_data.dart:102]: Nit: document what this class is for.

**Review by bkonyi — APPROVED** [review]
LGTM with nits.

Thanks for doing this!

File: packages/devtools_app/lib/src/shared/table_data.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/table_data.dart]: done

File: packages/devtools_app/lib/src/shared/table_data.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/table_data.dart:108]: done

**Pull request merged**